index.md

```diff
@@ -0,0 +1,791 @@
+
+```
+package com.example.directusapp.network
+
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+interface DirectusApiService {
+    companion object {
+        private const val BASE_URL = "https://directus.example.com"
+
+        fun create(): DirectusApiService {
+            val retrofit = Retrofit.Builder()
+                .baseUrl(BASE_URL)
+                .addConverterFactory(GsonConverterFactory.create())
+                .build()
+            return retrofit.create(DirectusApiService::class.java)
+        }
+    }
+}
+```
+
+The above code defines a `DirectusAPIService` that includes a `create()` function to set up a Retrofit instance. This function creates a `Retrofit.Builder` object, sets the base URL using `baseUrl(BASE_URL)`, adds the GsonConverterFactory for handling JSON data conversion, builds the Retrofit instance with `build()`, and creates an implementation of the DirectusApiService interface using `create(DirectusApiService::class.java)`.
+
+Similarly to the network package, crate a model and create a new Kotlin file named `Models` in the model package and define the app models:
+
+```
+package com.example.directusapp.model
+
+data class Author(
+    val id: Int,
+    val name: String,
+)
+
+data class Blog(
+    val id: Int,
+    val title: String,
+    val content: String,
+    val dateCreated: String,
+    val author: Author
+)
+
+data class Page(
+    val slug: String,
+    val title: String,
+    val content: String,
+)
+
+data class Global(
+    val id: Int,
+    val title: String,
+    val description: String,
+)
+
+data class BlogResponse(
+    val data: Blog
+)
+
+data class BlogsResponse(
+    val data: List<Blog>
+)
+
+data class PageResponse(
+    val data: List<Page>
+)
+
+data class GlobalResponse(
+    val data: Global
+)
+```
+
+The above code defines data classes for different Directus collections and their respective response models.
+
+## Using Global Metadata and Settings
+In your Directus Data Studio, click on Settings -> Data Model and create a new collection named `global`. Select 'Treat as a single object' under the Singleton option because this will only have a single entry containing the app's global metadata. Create two text input fields - one with the key `title` and one with `description`.
+
+Dirctus collections are not accessible to the public by default, click on Settings -> Access Control -> Public and give Read access to the Global collection.
+
+Then click on the content module and select the global collection. A collection would normally display a list of items, but since this is a singleton, it will launch directly into the one-item form. Enter information in the title and description field and hit save.
+
+![Creating global collection](./global.png)
+
+Update the code in your `DirectusHelper.kt` file in your network package to define a Get endpoint to fetch the global metadata from Directus:
```

Comment:
````suggestion
## Using Global Metadata and Settings

In your Directus Data Studio, click on **Settings -> Data Model** and create a new collection named `global`. Select 'Treat as a single object' under the Singleton option because this will only have a single entry containing the app's global metadata. Create two text input fields - one with the key `title` and one with `description`.

Dirctus collections are not accessible to the public by default, click on **Settings -> Access Control -> Public** and give **Read** access to the `global` collection.

Then click on the content module and select the global collection. A collection would normally display a list of items, but since this is a singleton, it will launch directly into the one-item form. Enter information in the title and description field and hit save.

![Creating global collection](./global.png)

Update the code in your `DirectusHelper.kt` file in your network package to define a Get endpoint to fetch the global metadata from Directus:
````